core/controllers/suggestion_test.py

```diff
@@ -0,0 +1,220 @@
+
+from constants import constants
+from core.domain import exp_domain
+from core.domain import exp_services
+from core.domain import rights_manager
+from core.domain import user_services
+from core.platform import models
+from core.tests import test_utils
+import feconf
+
+(suggestion_models, feedback_models) = models.Registry.import_models([
+    models.NAMES.suggestion, models.NAMES.feedback])
+
+
+class SuggestionUnitTests(test_utils.GenericTestBase):
+
+    EXP_ID = 'exp1'
+    TRANSLATION_LANGUAGE_CODE = 'en'
+
+    AUTHOR_EMAIL = 'author@example.com'
+    AUTHOR_EMAIL_2 = 'author2@example.com'
+    REVIEWER_EMAIL = 'reviewer@example.com'
+    ASSIGNED_REVIEWER_EMAIL = 'assigned_reviewer@example.com'
+
+    def setUp(self):
+        super(SuggestionUnitTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.signup(self.AUTHOR_EMAIL, 'author')
+        self.signup(self.AUTHOR_EMAIL_2, 'author2')
+        self.signup(self.ASSIGNED_REVIEWER_EMAIL, 'assignedReviewer')
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+        self.author_id = self.get_user_id_from_email(self.AUTHOR_EMAIL)
+        self.author_id_2 = self.get_user_id_from_email(self.AUTHOR_EMAIL_2)
+        self.reviewer_id = self.editor_id
+        self.assigned_reviewer_id = self.get_user_id_from_email(
+            self.ASSIGNED_REVIEWER_EMAIL)
+
+        self.editor = user_services.UserActionsInfo(self.editor_id)
+
+        # Login and create exploration and suggestions.
+        self.login(self.EDITOR_EMAIL)
+        exploration = (
+            self.save_new_linear_exp_with_state_names_and_interactions(
+                self.EXP_ID, self.editor_id, ['State 1', 'State 2', 'State 3'],
+                ['TextInput'], category='Algebra'))
+
+        self.old_content = exp_domain.SubtitledHtml('old content', {
+            self.TRANSLATION_LANGUAGE_CODE: exp_domain.AudioTranslation(
+                'filename.mp3', 20, False)
+        }).to_dict()
+
+        exploration.states['State 1'].update_content(self.old_content)
+        exploration.states['State 2'].update_content(self.old_content)
+        exploration.states['State 3'].update_content(self.old_content)
+        exp_services._save_exploration(self.editor_id, exploration, '', [])  # pylint: disable=protected-access
+
+        rights_manager.publish_exploration(self.editor, self.EXP_ID)
+        rights_manager.assign_role_for_exploration(
+            self.editor, self.EXP_ID, self.owner_id,
+            rights_manager.ROLE_EDITOR)
+
+        self.new_content = exp_domain.SubtitledHtml('new content', {
+            self.TRANSLATION_LANGUAGE_CODE: exp_domain.AudioTranslation(
+                'filename.mp3', 20, False)
+        }).to_dict()
+
+        self.logout()
+
+        if constants.USE_NEW_SUGGESTION_FRAMEWORK:
```

Comment:
How is this going to be tested if USE_NEW_SUGGESTION_FRAMEWORK is false?

Reply:
@nithusha21 -- typically there should not be "if" statements in tests. Use `with self.swap(...)` instead if you want to simulate various configurations of feconf constants.

Reply:
Done